import {
  testGetUserProfile,
  testGetUserProfileFail,
} from './mockGetUserProfile';
import { testLoginFail, testLoginSuccess } from './mockLogin';
import { testLogout } from './mockLogout';

const runMockTest = async (): Promise<void> => {
  try {
    console.log('‚úÖ Testing LOGIN with correct email and password...');
    const loginResponse = await testLoginSuccess();
    console.log('‚úÖ Login Success:', loginResponse);

    console.log('‚úÖ Testing GET USER PROFILE...');
    const userProfile = await testGetUserProfile(loginResponse.token);
    console.log('‚úÖ User Profile:', userProfile);

    console.log('‚úÖ Testing LOGOUT...');
    const logoutResponse = await testLogout();
    console.log('‚úÖ Logout Success:', logoutResponse);

    console.log('\nüö® Testing LOGIN with wrong password...');
    await testLoginFail();
  } catch (err) {
    if (err instanceof Error) {
      console.error('‚ùå ERROR:', err.message);
    } else {
      console.error('‚ùå ERROR:', err);
    }
  }

  try {
    console.log('\nüö® Testing GET USER PROFILE with invalid token...');
    await testGetUserProfileFail();
  } catch (err) {
    if (err instanceof Error) {
      console.error('‚ùå ERROR:', err.message);
    } else {
      console.error('‚ùå ERROR:', err);
    }
  }
};

runMockTest();